main_loop_tf/main.py

```diff
@@ -324,53 +325,54 @@ def __run(build_model):
                                              staircase=cfg.staircase)
         else:
             raise NotImplementedError()
-
         cfg.Optimizer = cfg.Optimizer(learning_rate=lr, **cfg.optimizer_params)
 
-        placeholders = [inputs_per_gpu, labels_per_gpu, prev_err]
-        val_placeholders = [val_inputs_per_gpu, labels_per_gpu]
-
-        # When the set size is not a multiple of the batch_size, the
-        # last batch will be smaller than batch_size. When this happen
-        # we might not be able to feed all the GPUs with an input for
-        # this batch. Here we compute the number of GPUs to use for the
-        # last batch of each set
-        n_gpus_per_set = {}
-        for s in ['train'] + cfg.val_on_sets:
-            if s == 'train':
-                params = cfg.dataset_params
-            else:
-                params = cfg.valid_params
-            d_params = copy.deepcopy(params)
-            d_params['batch_size'] *= cfg.num_splits
-            this_set = cfg.Dataset(which_set=s, **d_params)
-            batch_size = params['batch_size']
+        # Check if the last batch will not be processed by all the devices.
+        # When the number of samples of the set is not a multiple of the
+        # batch_size, the last batch will be smaller than batch_size.
+        # When this happens we might not be able to feed all the
+        # CPUs/GPUs with the last batch. Here, for each set, we compute
+        # the number of splits for the last batch
+        small_num_splits = {}
+        for s in ['train'] + ['eval_' + v for v in cfg.val_on_sets]:
+            params = cfg.dataset_params if s == 'train' else cfg.valid_params
+            this_set = cfg.Dataset(which_set=s, **params)
```

Comment:
The problem here is that the "eval_" prefix does not allow the dataset loader to match some set. 
Could be passing s.split("_")[1] to which_set a possible solution?